apps/cowswap-frontend/src/modules/swap/containers/SwapConfirmModal/index.tsx

```diff
@@ -49,11 +56,142 @@ export interface SwapConfirmModalProps {
   recipient?: string | null
 }
 
-// TODO: Break down this large function into smaller functions
-// TODO: Add proper return type annotation
-// eslint-disable-next-line max-lines-per-function, @typescript-eslint/explicit-function-return-type
-export function SwapConfirmModal(props: SwapConfirmModalProps) {
-  const { inputCurrencyInfo, outputCurrencyInfo, priceImpact, recipient, doTrade } = props
+interface LabelsAndTooltips {
+  slippageLabel?: React.ReactNode
+  slippageTooltip?: React.ReactNode
+  expectReceiveLabel?: React.ReactNode
+  minReceivedLabel?: React.ReactNode
+  minReceivedTooltip?: React.ReactNode
+  networkCostsSuffix?: React.ReactNode
+  networkCostsTooltipSuffix?: React.ReactNode
+}
+
+interface SwapConfirmModalData {
+  account: string | undefined
+  ensName: string | undefined
+  appData: AppDataInfo | null
+  receiveAmountInfo: ReceiveAmountInfo | null
+  tradeConfirmActions: TradeConfirmActions
+  slippage: Percent | null
+  deadline: number
+  shouldDisplayBridgeDetails: boolean
+  isQuoteLoading: boolean
+  bridgeProvider: BridgeProviderInfo | undefined
+  bridgeQuoteAmounts: BridgeQuoteAmounts | null
+  swapContext: QuoteSwapContext | null
+  bridgeContext: QuoteBridgeContext | null
+  rateInfoParams: RateInfoParams
+  submittedContent: (onDismiss: Command) => ReactElement
+  labelsAndTooltips: LabelsAndTooltips
+}
+
+function checkBalanceSufficiency(inputCurrencyInfo: CurrencyPreviewInfo, balances: Record<string, string>): boolean {
+  const current = inputCurrencyInfo?.amount?.currency
+
+  if (!current) {
+    return false
+  }
+
+  const normalisedAddress = getCurrencyAddress(current).toLowerCase()
+  const balance = balances[normalisedAddress]
+
+  if (!balance) {
+    return false
+  }
+
+  const balanceAsCurrencyAmount = CurrencyAmount.fromRawAmount(current, balance)
+
+  return Boolean(
+    inputCurrencyInfo?.amount?.equalTo(balanceAsCurrencyAmount) ||
+      inputCurrencyInfo?.amount?.lessThan(balanceAsCurrencyAmount),
+  )
+}
+
+function useDisableConfirmLogic(
```

Comment:
Would be great to keep the container file clean and move hooks to a separate file